template.yml

```diff
@@ -276,8 +279,74 @@ Resources:
       AutoPublishAlias: live
       ProvisionedConcurrencyConfig:
         ProvisionedConcurrentExecutions: !Ref ProvisionedConcurrentExecutions
-      Description: Compile and execute a JavaLab project.
+      Description: Compile and execute a JavaLab Theater project.
       MemorySize: 1769
+      Timeout: 300 # 90 -> TODO: Set this to 90 once the code-dot-org repo has updated
+      Role:
+        Fn::ImportValue: JavabuilderLambdaExecutionRole
+      Environment:
+        Variables:
+          AWS_LAMBDA_EXEC_WRAPPER: /opt/change_runtime_directory
+          OUTPUT_BUCKET_NAME: !Ref OutputBucket
+          GET_OUTPUT_URL: !Sub "https://${OutputDomain}"
+  BuildAndRunJavaNeighborhoodProjectFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      Layers:
+        - !Ref FontConfigurationLayer
+        - !Ref ChangeJavaRuntimeDirectoryLayer
+      Handler: org.code.javabuilder.LambdaRequestHandler::handleRequest
+      Runtime: java11
+      CodeUri: org-code-javabuilder/lib/build/distributions/lib.zip
+      AutoPublishAlias: live
+      ProvisionedConcurrencyConfig:
+        ProvisionedConcurrentExecutions: !Ref ProvisionedConcurrentExecutions
+      Description: Compile and execute a JavaLab Neighborhood project.
+      MemorySize: 512
+      Timeout: 60
+      Role:
+        Fn::ImportValue: JavabuilderLambdaExecutionRole
+      Environment:
+        Variables:
+          AWS_LAMBDA_EXEC_WRAPPER: /opt/change_runtime_directory
+          OUTPUT_BUCKET_NAME: !Ref OutputBucket
+          GET_OUTPUT_URL: !Sub "https://${OutputDomain}"
+  BuildAndRunJavaConsoleProjectFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      Layers:
+        - !Ref FontConfigurationLayer
+        - !Ref ChangeJavaRuntimeDirectoryLayer
+      Handler: org.code.javabuilder.LambdaRequestHandler::handleRequest
+      Runtime: java11
+      CodeUri: org-code-javabuilder/lib/build/distributions/lib.zip
+      AutoPublishAlias: live
+      ProvisionedConcurrencyConfig:
+        ProvisionedConcurrentExecutions: !Ref ProvisionedConcurrentExecutions
+      Description: Compile and execute a JavaLab Console project.
+      MemorySize: 512
+      Timeout: 60
+      Role:
+        Fn::ImportValue: JavabuilderLambdaExecutionRole
+      Environment:
+        Variables:
+          AWS_LAMBDA_EXEC_WRAPPER: /opt/change_runtime_directory
+          OUTPUT_BUCKET_NAME: !Ref OutputBucket
+          GET_OUTPUT_URL: !Sub "https://${OutputDomain}"
```

Comment:
This duplication is a little harder to avoid. Might be a good place for a [Cloudformation Mapping](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/mappings-section-structure.html), so you can signal to the reader that memory type and timeout are the interesting and likely-to-be-configured bits.

Reply:
I also realized I might be able to use a pattern similar to this "each" pattern: https://github.com/code-dot-org/code-dot-org/blob/004deda4d0d5408904a76eaec5c1345f190abbde/aws/cloudformation/components/alarms.yml.erb#L3

Reply:
Hah I just realized that's a .yml.erb.

Reply:
Actually, that gives me an idea, is it bad practice to use that mapping for all the fields that are the same? i.e. then it's an "update in one place" change whenever we update one of the shared fields.

Reply:
Could look something like this:
```
Mappings: 
  SharedLambdaSettings: 
    Handler: org.code.javabuilder.LambdaRequestHandler::handleRequest
    Runtime: java11
    CodeUri: org-code-javabuilder/lib/build/distributions/lib.zip
    AutoPublishAlias: live
    ...
  UniqueLambdaSettings
    Playground:
      MemorySize: 512
      Timeout: 60
...
      Handler: !FindInMap [SharedLambdaSettings, Handler]
      Runtime: !FindInMap [SharedLambdaSettings, Runtime]
      CodeUri: !FindInMap [SharedLambdaSettings, CodeUri]
      AutoPublishAlias: !FindInMap [SharedLambdaSettings, AutoPublishAlias]
      MemorySize: !FindInMap [UniqueLambdaSettings, Playground, MemorySize]
      Timeout: !FindInMap [UniqueLambdaSettings, Playground, Timeout]
```

Reply:
resolving this thread because the ERB solution looks great. As much as I don't want to overcomplicate it by bringing in another layer, it comes out pretty clean here.